app/src/main/java/com/pvp/app/ui/screen/friends/FriendsViewModel.kt

```diff
@@ -0,0 +1,161 @@
+package com.pvp.app.ui.screen.friends
+
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.FriendService
+import com.pvp.app.api.UserService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class FriendsViewModel @Inject constructor(
+    private val friendService: FriendService,
+    private val userService: UserService
+) : ViewModel() {
+
+    val toastMessage = mutableStateOf<String?>(null)
+
+    private val _isRequestSent = MutableStateFlow(false)
+    val isRequestSent = _isRequestSent.asStateFlow()
+
+    val user = userService.user.stateIn(
+        viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = null
+    )
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val friendObject = user.flatMapLatest { user ->
+        user?.email?.let { email ->
+            friendService.create(email)
```

Comment:
Every time user changes, we are still trying to create a friend object. To me it looks that getting nullable user in this ViewModel is hardly possible, since the route is locked until `User` object becomes not-null. So you could just call `user.first()!!` or `user.first()`, but make `Flow<User?>` to filterNotNull(mapNotNull?) and then create its friend object within ViewModel's init block (or other private function, but the idea is that it would be called in the init block).

However. I think it would be best to create the friend object whenever a new user is created after register. What you are doing now is probably to support old users so that they get their objects created. Well, you could write a small piece of code that would go through every user and create their friend objects upon application start, then remove that piece of code and leave it all for the register process by modifying it a bit.

You choose how you want to handle this, if you want, even leave this thing as it is, it's not like it takes 1 or more seconds to complete. But still, small resource waste, especially if we had to pay for our database reads. 😄

Also, two different purpose methods should be separated.